server/src/main/java/org/elasticsearch/search/aggregations/bucket/filter/QueryToFilterAdapter.java

```diff
@@ -82,18 +81,6 @@
      * {@link #weight()} to build it when needed.
      */
     private Weight weight;
-    /**
-     * Scorer for each segment or {@code null} if we haven't built the scorer.
-     * Use {@link #bulkScorer(LeafReaderContext, Runnable)} to build the scorer
-     * when needed.
-     */
-    private BulkScorer[] bulkScorers;
```

Comment:
No need to cache scorers because we use them once. We still want to cache the weight because we use it for every segment.